### ISSUE TITLE
Provide example for HA singleton service

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#77]

**Comment by tdiesler** [comment]
Currently getting this with 9.0.2

```
java.lang.IllegalStateException: java.lang.UnsupportedOperationException
	at java.util.Collections$1.remove(Collections.java:4684)
	at org.wildfly.clustering.server.singleton.SingletonService$SingletonDispatcher.getValueRef(SingletonService.java:266)
	at org.wildfly.clustering.server.singleton.SingletonService.getValue(SingletonService.java:225)
	at org.wildfly.clustering.server.singleton.SingletonService.getValue(SingletonService.java:69)
	at org.wildfly.clustering.service.AsynchronousServiceBuilder.getValue(AsynchronousServiceBuilder.java:92)
	at org.jboss.msc.service.ServiceControllerImpl.awaitValue(ServiceControllerImpl.java:1166)
	at org.wildfly.camel.test.singleton.SingletonRouteTest.testSingletonService(SingletonRouteTest.java:138)
```

CrossRef: https://issues.jboss.org/browse/WFLY-5612

**Comment by tdiesler** [comment]
Blocked by #958

**Comment by tdiesler** [comment]
Won't fix - see #1045
